mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/CassandraMessageManager.java

```diff
@@ -80,4 +81,12 @@ protected void storeAttachment(final MailboxMessage message, final List<MessageA
                     .collect(Guavate.toImmutableList()),
                 message.getMessageId());
     }
+
+    @Override
+    protected MailboxMessage copyMessage(MailboxMessage message) throws MailboxException {
```

Comment:
that's clone, no?

Reply:
From Effective Java 3rd Edition : 

(About Clone in ITEM 13: OVERRIDE CLONE JUDICIOUSLY)

> Is all this complexity really necessary? Rarely. If you extend a class that
already implements Cloneable , you have little choice but to implement a well-
behaved clone method. Otherwise, you are usually better off providing an
alternative means of object copying. A better approach to object copying is to
provide a copy constructor or copy factory. A copy constructor is simply a
constructor that takes a single argument whose type is the class containing the
constructor, for example,